pkg/storagepool/storagepool.go

```diff
@@ -162,3 +165,97 @@ func (v *casPoolOperation) Delete() (*v1alpha1.CasPool, error) {
 	}
 	return pool, nil
 }
+
+type storagePoolOperation struct {
+	poolName  string
+	k8sClient *m_k8s_client.K8sClient
+}
+
+func NewStoragePoolOperation(poolName string) (*storagePoolOperation, error) {
+	kc, err := m_k8s_client.NewK8sClient("")
+	if err != nil {
+		return nil, err
+	}
+	// Put pool object inside casPoolOperation object
+	return &storagePoolOperation{
+		poolName:  poolName,
+		k8sClient: kc,
+	}, err
+}
+
+func (s *storagePoolOperation) List() (*v1alpha1.StoragePoolList, error) {
+	if s.k8sClient == nil {
+		return nil, fmt.Errorf("Unable to fetch K8s client")
```

Comment:
First letter should be small in error msg

Reply:
fixed